modules/caddyhttp/fileserver/caddyfile.go

```diff
@@ -33,8 +33,43 @@ func init() {
 	httpcaddyfile.RegisterDirective("try_files", parseTryFiles)
 }
 
-// parseCaddyfile parses the file_server directive. It enables the static file
-// server and configures it with this syntax:
+// parseCaddyfile parses the file_server directive.
+// See UnmarshalCaddyfile for the syntax.
+func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error) {
+	fsrv := new(FileServer)
+	err := fsrv.UnmarshalCaddyfile(h.Dispenser)
+	if err != nil {
+		return fsrv, err
+	}
+
+	// Hide the Caddyfile (and any imported Caddyfiles).
+	// This needs to be done in here instead of UnmarshalCaddyfile
+	// because UnmarshalCaddyfile only has access to the dispenser
+	// and not the helper, and only the helper has access to the
+	// Caddyfiles function. This does mean that Caddyfiles will
+	// only be hidden when the directive is used directly, and not
+	// if used in some embedded fashion (e.g. via another directive
+	// as a shortcut).
```

Comment:
This could be a big gotcha...

I'm not sure what to do about this, currently. Hmm.

Reply:
We could move that logic to a method of the `FileServer` module, taking `h` as input, then embedders may call that method to add the `hide` stuff on top if necessary. We do the same kind of thing with `reverse_proxy` i.e. `rp.FinalizeUnmarshalCaddyfile(h)`, we could make a similar `Finalize` for `file_server`.

Reply:
Did that :+1:

Reply:
Oh -- good idea 🤔 Why didn't I think of that...